docs/source/ProgrammingGuide/Initialization.md

```diff
@@ -43,66 +45,68 @@ Kokkos chooses the two spaces using the following list:
 
-## 5.2 Initialization by struct
+## 5.2 Initialization by environment variable
 
-Instead of giving [`Kokkos::initialize()`](../API/core/initialize_finalize/initialize) command-line arguments, one may directly pass in initialization parameters using the following struct:
+Instead of using command-line arguments, one may use environment variables. The environment variables are identical to the arguments in Table 5.1 but they are upper case and the dash is replaced by an underscore. For example, if we want to set the number of threads to 3, we have
+
+  KOKKOS_NUM_THREADS=3
+
+## 5.3 Initialization by struct
+
+Instead of giving [`Kokkos::initialize()`](../API/core/initialize_finalize/initialize) command-line arguments, one may directly pass in initialization parameters using the `Kokkos::InitializationSettings` struct.  If one wants to set a options using the struct, one can use the set functions `set_xxx` where `xxx` is the identical to the arguments in Table 5.1 where the dash has been replaced by an unsderscore. To check if a variable has been set, one can use the `has_xxx` functions. Finally, to get the value that was set, one can use the `get_xxx` functions.
 
-```c++
-struct Kokkos::InitArguments {
-  int num_threads;
-  int num_numa;
-  int device_id;
-  int ndevices;
-  int skip_device;
-  bool disable_warnings;
-};
-```
-The `num_threads` field corresponds to the `--kokkos-threads` command-line argument, `num_numa` to `--kokkos-numa`, `device_id` to `--device-id`, `ndevices` to the first value in `--num-devices` and `skip_devices` to the second value in `--num-devices`. (See Table 5.1 for details.) Not all parameters are observed by all execution spaces, and the struct might expand in the future if needed.
 
-If you set `num_threads` or `num_numa` to zero or less, Kokkos will try to determine default values if possible or otherwise set them to 1. In particular, Kokkos can use the `hwloc` library to determine default settings using the assumption that the process binding mask is unique, i.e., that this process does not share any cores with another process. Note that the default value of each parameter is -1.
+If you do not set `num_threads`, Kokkos will try to determine a default value if possible or otherwise set it to 1. In particular, Kokkos can use the `hwloc` library to determine default settings using the assumption that the process binding mask is unique, i.e., that this process does not share any cores with another process. Note that the default value of each parameter is -1.
 
 Here is an example of how to use the struct.
 
 ```c++
-Kokkos::InitArguments args;
-// 8 (CPU) threads per NUMA region
-args.num_threads = 8;
-// 2 (CPU) NUMA regions per process
-args.num_numa = 2;
+Kokkos::InitializationSettings settings;
+// 8 (CPU) threads
+settinge.set_num_threads(8);
 // If Kokkos was built with CUDA enabled, use the GPU with device ID 1.
-args.device_id = 1;
+settings.set_device_id(1);
 
```

Comment:
Is this behavior the same in other backends?  Either way, we should note the similarities or differences.